truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -109,22 +168,25 @@ public RubyBasicObject initialize(RubyBasicObject self, RubyBasicObject v) {
         }
 
         @Specialization(guards = "isRubyString(v)")
+        @CompilerDirectives.TruffleBoundary
         public RubyBasicObject initializeFromString(RubyBasicObject self, RubyBasicObject v) {
-            // TODO (pitr 20-May-2015): add NaN, Infinity handling
+            // TODO (pitr 21-May-2015): add profile?
+            // TODO (pitr 25-May-2015): construction of positive and negative zero
             switch (v.toString()) {
```

Comment:
`RubyString#toString` is the exception to that note. What we don't want to do is things like `to_s` as `toString` or `hash` as `hatched` - we don't want to map the Ruby API into the Java API like that (because those methods don't give us anywhere to store state for optimisation).